### DISCUSSION TITLE
SuperAdmin can not match rules anymore

### DISCUSSION THREAD

**Discussion by ghost** [Keyfactor/ejbca-ce#305] [Q&A] [answered]
Hello,

With my [previous issue fixed](https://github.com/Keyfactor/ejbca-ce/discussions/302), I successfully deployed EJBCA with docker compose along with NGINX. I needed to do the steps in specific order.

However, I do not know why, after tweaking nginx, I did a `docker compose down & docker compose up -d` and now my SuperAdmin cert does not have access anymore to /ejbca/adminweb, everything is still in the database but Edge, Firefox or Chrome says `Authentication failed for certificate with no access`

and the logs tells me:

```
ejbca           | 2023-06-20 14:53:50,130+0000 INFO  [org.cesecore.audit.impl.log4j.Log4jDevice] (default task-1) 2023-06-20 14:53:50+00:00;ADMINWEB_ADMINISTRATORLOGGEDIN;FAILURE;ADMINWEB;EJBCA;192.168.1.108 (TRANSPORT_PLAIN);;;;remoteip=192.168.1.108;reason=AuthenticationToken has no access
```

My Certificate
![image](https://github.com/Keyfactor/ejbca-ce/assets/92848369/e2a1de59-bfaf-432c-b62c-5c6383e5126a)

EJBCA rules:

```
docker exec -it ejbca ejbca.sh roles listadmins --role 'Super Administrator Role'
2023-06-20 14:53:38,579+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) 'ManagementCA' WITH_COMMONNAME TYPE_EQUALCASE "SuperAdmin" ""
2023-06-20 14:53:38,598+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) 'ManagementCA' WITH_SERIALNUMBER TYPE_EQUALCASEINS "31A0E35FA7FA9EB90A15FCCA690FE97E27C63AF5" "SuperAdmin cert serial"
2023-06-20 14:53:38,610+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) 'ManagementCA' WITH_SERIALNUMBER TYPE_EQUALCASE "F4F8F227CCE11A85E300CE2807AEC7F43A454F0" ""
2023-06-20 14:53:38,625+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) 'ManagementCA' WITH_SERIALNUMBER TYPE_EQUALCASE "551DE1A21040BCE9D598AA5DBC97BDD0BED197CF" ""
2023-06-20 14:53:38,625+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) [Admin not bound to CA or provider] USERNAME TYPE_EQUALCASE "ejbca" ""
2023-06-20 14:53:38,625+0000 INFO  [org.ejbca.ui.cli.roles.ListAdminsCommand] (main) [Admin not bound to CA or provider] TRANSPORT_CONFIDENTIAL TYPE_UNUSED "" "Initial RoleMember."
```

- I already tried to generate a new one: 

```
docker exec -it ejbca ejbca.sh ra setendentitystatus superadmin 10
docker exec -it ejbca ejbca.sh ra setclearpwd superadmin foo123
docker exec -it ejbca ejbca.sh batch
```

and download the .p12 to my cert store and adding the new serial rules with no success

Any help regarding this strange issue ?

**Comment by ghost** [comment]
Same for NGINX it self,

I added just to test

```
ssl_client_certificate /etc/nginx/certs/management_CA.pem;
   ssl_verify_client optional;


   location / {

   if ($ssl_client_verify != SUCCESS) {
     return 403;
   }
```

And now I can not access the public web, it seems even nginx does not recognize my SuperAdmin cert.
I comment for now the if condition and have access to the public web. But it seems NGINX does not send my client cert to EJBCA to confirm I can access.

Is it because NGINX provide frontend with TLS Certificate signed by COMPANY SUB CA and that my Superadmin is signed by Management CA (one guy had this [issue](https://sourceforge.net/p/ejbca/discussion/123123/thread/d7b4c6b884/#a6b7/9ada/8100/4478)) in 2019.

**Reply by primetomas** [reply]
Yes almost. That is how TLS client cert authentication works. The server have a set of "trusted client root CAs" configured. The CA of the superadmin client cert must be included in this list, which is sent to the client to determine which certificates it can send.

**Reply by ghost** [reply]
Well, I will try with an httpd container instead and see if it works.

**Reply by primetomas** [reply]
You can condfigure this somewhere in nginx, you simply need to add ManagementCA.pem in the right place in the nginx TLS configuration.

**Reply by ghost** [reply]
Is [this confirm](https://sourceforge.net/p/ejbca/discussion/123123/thread/b37f58cf33/#542a) that NGINX can not forward TLS-SSL-client-cert to Apache as it will not understand the header format ?

EDIT: it does not work for traefik in this thread, I misread.

**Reply by ghost** [reply, marked as answer]
I'm using this with NGINX:

```
ssl_client_certificate /etc/nginx/certs/management_CA.pem;
   ssl_verify_client optional;


   location / {

   #if ($ssl_client_verify != SUCCESS) {
    # return 403;
   #}

        proxy_pass                              http://ejbca-node1:8082/;
        proxy_set_header Host                   $http_host;
        proxy_set_header X-Real-IP              $remote_addr;
        proxy_set_header X-Forwarded-For        $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto      $scheme;
        proxy_set_header SSL_CLIENT_CERT        $ssl_client_cert;
    }

```

And yet this does not work.

EDIT: Well, i'm a bit ashamed... 

The problem was that I did not clear the cookies and cache neither closing Chrome since a long time ago.

It works with the same settings.
